guide/students/student_management_service.bal

```diff
@@ -0,0 +1,336 @@
+        //sending a request to mysql endpoint and getting a response with required data table
+        _ = observe:finishSpan(spanId2);
+
+        table<Student> dt;
+        // a table is declared with Student as its type
+        //match operator used to check if the response returned value with one of the types below
+        match selectRet {
+            table tableReturned => dt = tableReturned;
+            error e => io:println("Select data from student table failed: "
+                    + e.message);
+        }
+
+        //student details displayed on server side for reference purpose
+        io:println("Iterating data first time:");
+        foreach row in dt {
+            io:println("Student:" + row.id + "|" + row.name + "|" + row.age);
+        }
+
+        // table is converted to json
+        var jsonConversionRet = <json>dt;
+        match jsonConversionRet {
+            json jsonRes => {
+                status = jsonRes;
+            }
+            error e => {
+                status = { "Status": "Data Not available", "Error": e.message };
+            }
+        }
+        //Sending back the converted json data to the request made to this service
+        response.setJsonPayload(untaint status);
+        _ = httpConnection->respond(response);
+
+        _ = observe:finishSpan(chSpanId);
+        _ = observe:addTagToSpan(spanId = -1, "tot_requests", <string>req_count);
+        _ = observe:addTagToSpan(spanId = -1, "error_counts", <string>errors);
+
+    }
+
+    @http:ResourceConfig {
+        methods: ["GET"],
+        path: "/testError"
+    }
+    //viewStudent service to get all the students details and send to the requested user
+    testError(endpoint httpConnection, http:Request request) {
+        req_count++;
+        http:Response response;
+
+        errors++;
+        io:println(errors);
+        _ = observe:addTagToSpan(spanId = -1, "error_counts", <string>errors);
+        _ = observe:addTagToSpan(spanId = -1, "tot_requests", <string>req_count);
+        log:printError("error test");
+        response.setTextPayload("Test Error made");
+        _ = httpConnection->respond(response);
+    }
+
+    @http:ResourceConfig {
+        methods: ["GET"],
+        path: "/deleteStu/{stuId}"
+    }
+    //deleteStudents service for deleteing a student using id
+    deleteStudent(endpoint httpConnection, http:Request request, int stuId) {
+        req_count++;
+        http:Response response;
+        json status = {};
+
+        //calling deleteData function with id as parameter and get a return json object
+        var ret = deleteData(stuId);
+        io:println(ret);
+        //Pass the obtained json object to the request
+        response.setJsonPayload(ret);
+        _ = httpConnection->respond(response);
+        _ = observe:addTagToSpan(spanId = -1, "tot_requests", <string>req_count);
+        _ = observe:addTagToSpan(spanId = -1, "error_counts", <string>errors);
```

Comment:
I think we can omit `spanId = -1` since -1 is the default value

Reply:
Yeah sure. I will do it.